src/preact/base-element.js

```diff
@@ -725,6 +743,23 @@ function collectProps(Ctor, element, ref, defaultProps) {
       devAssert(def.attrs);
       value = def.parseAttrs(element);
     }
+    if (def.attrPrefix) {
+      const currObj = {};
+      const attrs = element.attributes;
+      for (let i = 0; i < attrs.length; i++) {
+        const attrib = attrs[i];
+        if (matchesAttrPrefix(attrib.name, def.attrPrefix)) {
+          currObj[
+            dashToCamelCase(
+              attrib.name.substring(def.attrPrefix.length, attrib.name.length)
+            )
+          ] = attrib.value;
+        }
+      }
+      if (Object.keys(currObj).length > 0) {
```

Comment:
Nit: This is better done by storing a separate boolean value, doing `Object.keys` is expensive.